src/target/riscv/riscv-013.c

```diff
@@ -1023,6 +1033,23 @@ static int read_abstract_arg(struct target *target, riscv_reg_t *value,
 	return result;
 }
 
+static void abstract_data_write_fill_batch(struct riscv_batch *batch,
+		riscv_reg_t value, unsigned int index, unsigned int size_bits)
+{
+	assert(size_bits % 32 == 0);
+	const unsigned int size_in_words = size_bits / 32;
+	const unsigned int offset = index * size_in_words;
```

Comment:
If you like:

```suggestion
	const unsigned int offset = index * size_in_words;
	assert(value <= 0xffffffffu || size_in_words > 1);	
```

There are probably few other functions (not necessarily touched in this merge request) where the same assertion could be added.

Reply:
Addressed (only in this function).

I don't think similar assertion is necessary in `register_write_abstract()`, since it just passes `value` to `abstract_data_write_fill_batch()`.

As for other cases, I will keep in mind such assertions going forward.